### PR TITLE
W-13190334: Adapt the tracing layers to the overriding configuration.

### PR DESCRIPTION
**Pull request by alopezbizon**

### CODE REVIEW

**Comment by alopezbizon** [comment]
--validate

**Comment by alopezbizon** [comment]
--validate

File: modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/ExecutionInitialExportInfo.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ */
+package org.mule.runtime.tracer.customization.impl.export;
+
+import org.mule.runtime.api.component.Component;
+import org.mule.runtime.tracer.api.span.info.InitialExportInfo;
+import org.mule.runtime.tracer.customization.api.InitialExportInfoProvider;
+
+import java.util.Set;
+
+public class ExecutionInitialExportInfo implements InitialExportInfo {
```
Comment (fsgonz) [inline modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/ExecutionInitialExportInfo.java:15]: add jdoc.

File: modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/provider/DebugInitialExportInfoProvider.java [hunk 2]
```diff
@@ -33,10 +33,12 @@
 /**
  * An {@link InitialExportInfoProvider} corresponding to the {@link TracingLevel#DEBUG}
  *
- * @since 4.6.0
+ * @since 4.5.0
  */
 public class DebugInitialExportInfoProvider extends MonitoringInitialExportInfoProvider {
 
+  private boolean isOverride = false;
```
Comment (fsgonz) [inline modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/provider/DebugInitialExportInfoProvider.java:40]: Is this correct? The debug initialExportInfoProvider provides exportInfo which are the one retrieved for each location. It is used for every exportSpanInfo.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/TracingLevelExportInfo.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  private InitialExportInfo getInitialExportInfo() {
+    if (this.spanIdentifier instanceof Component) {
+      return this.initialExportInfoProvider.getInitialExportInfo((Component) spanIdentifier);
+    }
+    return this.initialExportInfoProvider.getInitialExportInfo((String) spanIdentifier);
+  }
+
+  public void propagateExportInfo(TracingLevelExportInfo tracingLevelExportInfo) {
```
Comment (sofiamorseletto) [inline modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/TracingLevelExportInfo.java:53]: Could we rename this to parentTracingLevelExportInfo? Just for clarity

Comment (alopezbizon) [reply modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/TracingLevelExportInfo.java]: Sure. It's done.

File: modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/ExecutionInitialExportInfo.java (hunk 1 above)
Comment (alopezbizon) [reply modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/export/ExecutionInitialExportInfo.java:19]: Done. Please review it.

File: modules/tracing/mule-tracer-configuration-api/src/main/java/org/mule/runtime/tracing/level/api/config/TracingLevelConfiguration.java [hunk 4]
```diff
@@ -7,16 +7,14 @@
...
  *
- * @since 4.6.0
+ * @since 4.5.0
  */
 public interface TracingLevelConfiguration {
 
   TracingLevel getTracingLevel();
 
-  Map<String, TracingLevel> getTracingLevelOverrides();
+  TracingLevel getTracingLevelOverride(String location);
```
[the remark this answers is not in the document]
Comment (alopezbizon) [reply modules/tracing/mule-tracer-configuration-api/src/main/java/org/mule/runtime/tracing/level/api/config/TracingLevelConfiguration.java:29]: Done. Please review it.

File: modules/tracing/mule-tracer-api/src/main/java/org/mule/runtime/tracer/api/span/info/InitialExportInfo.java [hunk 5]
```diff
@@ -58,4 +58,6 @@ default boolean isExportable() {
   default Set<String> noExportUntil() {
     return emptySet();
   }
+
+  default void propagateInitialExportInfo(InitialExportInfo initialExportInfo) {}
```
[the remark this answers is not in the document]
Comment (alopezbizon) [reply modules/tracing/mule-tracer-api/src/main/java/org/mule/runtime/tracer/api/span/info/InitialExportInfo.java:67]: Done. Please review it.

**Review by fsgonz — APPROVED** [review]

**Pull request merged**